guides/source/association_basics.md

```diff
@@ -956,6 +957,10 @@ end
 
 If you set the `:validate` option to `true`, then associated objects will be validated whenever you save this object. By default, this is `false`: associated objects will not be validated when this object is saved.
 
+##### `:optional`
+
+If you set the `:optional` option to `true`, then associated object will be validated for presence. By default, this is `false`: associated objects will be validated for presence.
```

Comment:
Ok, this was fixed in https://github.com/rails/rails/commit/803ef74f9bbcbc71af12060157a996f7b65e24db